### ISSUE TITLE
Importing deletions on 5'/3' staple end from Cadnano

### ISSUE THREAD

**Issue by mmguar** [UC-Davis-molecular-computing/scadnano#825]
I am importing a cadnano file that has several deletions that correspond to 5'-3' ends of staples.
From previous issues reported, this does not seem to be allowed by scadnano, which wants you to just design the staple one base shorter.
This is what it looks like:
![image](https://user-images.githubusercontent.com/56086936/198115503-ed152df6-0855-42ee-8ffe-52f0aa26f756.png)
I assume the balloon is telling me that there is something wrong there? Assigning a sequence does not work either, until I change the length of the staple.

Thus I was wondering if it would be possible to modify the candnano import to automatically fix these, as it can be ab it tedious by hand.

Thanks!

**Comment by dave-doty** [comment]
That's correct, in scadnano deletions and insertions can only appear "interior" to a domain, not on an end. So the import is ignoring the deletion on the 3' end, and since the other deletion on the scaffold is still present, it's warning you that there's an unpair deletion.

We could try to come up with something to fix in the import, but I won't be able to get to it soon unfortunately, so for now you'll have to do it by hand or with a Python script.

**Comment by dave-doty** [comment]
Ideally we have a more explicit warning, where if the import wants to get rid of a deletion, it warns the user or offers them the option to shift the deletion. (the latter could be a bit tricky to cover all the cases)

**Comment by dave-doty** [comment]
See also https://github.com/UC-Davis-molecular-computing/scadnano-python-package/issues/260
